test/integration/tests/3863-purge-command/Main.hs

```diff
@@ -2,15 +2,19 @@ import StackTest
 import Data.Maybe (listToMaybe, fromMaybe)
 import System.Directory
 import System.FilePath
+import Data.List (filter)
+
+newline :: Char
+newline = '\n'
 
 main :: IO ()
 main =
   -- For these commands, we'll need to know the `dist` directory.
   -- This is usually `.stack-work/dist/$compiler-variant/Cabal-xxxx`
-  stackCheckStdout [defaultResolverArg, "path", "--dist-dir"] $ \distDir ->
-
-    stackCheckStdout [defaultResolverArg, "path", "--local-install-root"] $ \localInstallRoot -> do
-
+  stackCheckStdout [defaultResolverArg, "path", "--dist-dir"] $ \distDir' -> do
+    let distDir = filter (\x -> x /= newline) distDir'
```

Comment:
Where are the newline characters occuring?

I can see how on windows there may be a problem, but surely the extraneous newline character would only appear at the end of the line?

Reply:
Never mind, I see what's going on… using the output of the `stack path` commands is problematic (by sheer coincidence an issue I've just encountered in another package). Rather than using the filter, though, this should just trim the final newline!